src/modules/Speech_Recognition/Whisper.py

```diff
@@ -1,18 +1,22 @@
 """Whisper Speech Recognition Module"""
 
-import whisperx
 import sys
 
-from modules.console_colors import (
-    ULTRASINGER_HEAD,
-    blue_highlighted,
-    red_highlighted,
-)
+import whisperx
+
+from modules.console_colors import ULTRASINGER_HEAD, blue_highlighted, red_highlighted
 from modules.Speech_Recognition.TranscribedData import TranscribedData
 
 
-def transcribe_with_whisper(audio_path: str, model: str, device="cpu", model_name: str = None) -> (
-list[TranscribedData], str):
+def transcribe_with_whisper(
```

Comment:
Would torch.cuda OutOfMemoryError also work here?

Reply:
I can't reproduce my own OOM errors, no I've tried upping the batch size to no avail.

These issues however lead me to believe it's the same error class: https://github.com/m-bain/whisperX/issues/388 https://github.com/m-bain/whisperX/issues/102 https://github.com/m-bain/whisperX/issues/315

So I suggest catching the same error and producing the same message for whisper as well.

Reply:
Check the diff, both errors are printed, an explanation of how to address it is printed afterwards, so it should be the first thing the user sees, and then the program exits.

Reply:
Nice!